### PR TITLE
[TextFields] Added tests for accessibilityLabel

### PR DESCRIPTION
**Pull request by randallli**
Related to https://github.com/material-components/material-components-ios/pull/8804 which got feedback asking for tests. This puts in tests before that change.

more info at #7513

### CODE REVIEW

File: components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (jverkoey) [inline components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m]: Copyright year.

File: components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+@interface MDCTextFieldAccessibilityTests : XCTestCase
+
+@end
+
+@implementation MDCTextFieldAccessibilityTests
+
+- (void)testUITextFieldAccessibilityLabel {
+  // Given
+  UITextField *field = [[UITextField alloc] init];
+  field.accessibilityLabel = @"blah";
```
Comment (jverkoey) [inline components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m]: Please move the assignments into a "when" section here and throughout.

File: components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  MDCTextField *field = [[MDCTextField alloc] init];
+  field.accessibilityLabel = nil;
+  field.placeholderLabel.accessibilityLabel = @"blah";
+  field.leadingUnderlineLabel.accessibilityLabel = @"bloog";
+  field.trailingUnderlineLabel.accessibilityLabel = @"flah";
+
+  // Then
+  XCTAssertEqualObjects(field.accessibilityLabel, @"blah, bloog, flah");
+}
+@end
```
Comment (jverkoey) [inline components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m:91]: nit: newline at end of file.

File: components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  // Given
+  MDCTextField *field = [[MDCTextField alloc] init];
+  field.accessibilityLabel = nil;
+  field.placeholderLabel.accessibilityLabel = @"blah";
+  field.leadingUnderlineLabel.accessibilityLabel = @"bloog";
+  field.trailingUnderlineLabel.accessibilityLabel = @"flah";
+
+  // Then
+  XCTAssertEqualObjects(field.accessibilityLabel, @"blah, bloog, flah");
+}
```
Comment (jverkoey) [inline components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m:89]: What is the expected behavior for the following "when"?

```objc
  field.accessibilityLabel = @"hai";
  field.placeholderLabel.accessibilityLabel = @"blah";
  field.leadingUnderlineLabel.accessibilityLabel = @"bloog";
  field.trailingUnderlineLabel.accessibilityLabel = @"flah";
```

Consider adding a test for this as well.

**Review by jverkoey — APPROVED** [review]

File: components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m (hunk 3 above)
Comment (randallli) [reply components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m:91]: I am not sure why GitHub is not rendering the last line here and in the [raw view](https://github.com/material-components/material-components-ios/blob/f11a6f9e4d9f95caf4ec0556745cd7a5f7f52865/components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m) but there are 92 lines in this file. 

The summary in the raw view shows 92 but only displays 91 lines.

File: components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m (hunk 4 above)
Comment (randallli) [reply components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m:89]: Added.

File: components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m (hunk 1 above)
Comment (randallli) [reply components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m]: Done.

File: components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m (hunk 2 above)
Comment (randallli) [reply components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m]: Done.

File: components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m (hunk 3 above)
Comment (randallli) [reply components/TextFields/tests/unit/MDCTextFieldAccessibilityTests.m:91]: (i assume you want an additional new line after `@end`)

**Pull request merged**